### ISSUE TITLE
Zed window never updated on startup

### ISSUE THREAD

**Issue by drdozer** [zed-industries/zed#19163]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When I launch zed, it creates a new window. It's registered on the task bar as "empty project". However, the window has no decorations and is a grab of whatever was visible in that region at the time zed was launched. It can't be killed without going to the terminal and using the kill command.

### Environment

$ zed -v
Zed 0.156.1 – /home/nmrp3/.local/zed.app/libexec/zed-editor
$ uname -a
Linux persephone 6.8.0-45-generic #45-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 30 12:02:04 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
$ zed --foreground
[2024-10-13T21:32:43+01:00 ERROR fs] No such file or directory (os error 2) about ["/home/nmrp3/.config/github-copilot"]
MESA-INTEL: warning: Haswell Vulkan support is incomplete
[2024-10-13T21:32:43+01:00 ERROR assistant] no worktrees when constructing LocalLspAdapterDelegate

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
What Linux distribution/version are you running? (Ubuntu 22? or an Ubuntu derivative)?
Which video card(s) and driver versions are you running?

**Comment by drdozer** [comment]
I'm applying the ppa now. Will let you know if that helps.

My laptop has an intel and nvidia card. I would not be surprised if it had reverted to running on the intel one, because that keeps happening after updates. I'm running stock ubuntu on this laptop. 

Operating System: Kubuntu 24.04
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.13
Kernel Version: 6.8.0-45-generic (64-bit)
Graphics Platform: X11
Processors: 8 × Intel® Core™ i7-4710MQ CPU @ 2.50GHz
Memory: 15.5 GiB of RAM
Graphics Processor: NVIDIA GeForce 840M/PCIe/SSE2
Manufacturer: Notebook
Product Name: W65_67SF
System Version: Not Applicable

**Comment by drdozer** [comment]
No. Tested out that ppa, and I get the same thing. Happy to try out other things if you can suggest them.

**Comment by drdozer** [comment]
I can't run the "copy system specs" action as I never get a functioning window to interact with.

**Comment by otonnesen** [comment]
Having the exact same issue with onboard intel graphics (Lenovo ThinkPad T14 Gen 5).

OS: Debian 12.7
Kernel: 6.10.6+bpo-amd64
Graphics Platform: X11
CPU: Intel(R) Core(TM) Ultra 7 155U

**Comment by notpeter** [comment]
Yeah, it looks like the [mesa in debian stable](https://tracker.debian.org/pkg/mesa) is 22 (22.3.6-1+deb 12u1) where that PPA that folks have reported fixing their issues provides mesa 24. Debian testing/unstable (Trixie) has 24, but the release date for that being Debian 13 stable hasn't been announced (likely summer 2025).

**Comment by otonnesen** [comment]
Good shout, I tried mesa 24 from backports and it's working fine now. Thanks!
